packages/playwright-core/src/server/injected/roleUtils.ts

```diff
@@ -590,6 +617,11 @@ function getElementAccessibleNameInternal(element: Element, options: AccessibleN
       if (skipSlotted && (node as Element | Text).assignedSlot)
         return;
       if (node.nodeType === 1 /* Node.ELEMENT_NODE */) {
+        // Do not contribute from selected roles.
+        const role = getAriaRole(node as Element);
+        if (role && doNotContributeToParentName.has(role))
```

Comment:
I think we should move this around line 612 instead. Otherwise, IIUC, we entirely skip children that have one of these roles and `aria-label` or `aria-labelledby`, while Blink does not.